### PR TITLE
Use join to get return value of threads

### PR DESCRIPTION
**Pull request by NilSet**
This is a work-around for mpsc channels not working. Still need to get them working to address redox-os/redox#542

### CODE REVIEW

File: src/rusthello/players/mod.rs [hunk 1]
```diff
@@ -15,7 +13,7 @@ const TIME_LIMIT: u32 = 8 * 100000000;
 const NUM_CELLS: u8 = ( reversi::BOARD_SIZE * reversi::BOARD_SIZE ) as u8;
 
 
-#[derive(Clone)]
+#[derive(Clone,Debug)]
```
Comment (ticki) [inline src/rusthello/players/mod.rs]: Use a space after `,`.

File: src/rusthello/players/mod.rs [hunk 2]
```diff
@@ -50,7 +48,7 @@ impl Score {
 
 
 
-#[derive(Clone)]
+#[derive(Clone,Debug)]
```
Comment (ticki) [inline src/rusthello/players/mod.rs]: Here too.

**Comment by ticki** [comment]
@EGhiorzi, you gotta review this one.

**Comment by ticki** [comment]
> This is a work-around for mpsc channels not working.

No, they do work. There is some problems with the scheduler, though.

**Comment by HenryTheCat** [comment]
It looks very good to me, it's a definite improvement (and I learned about `join`). Apart from @ticki's notes about the missing space, I have no other suggestion. Should I test the code or is it enough to look at it (sorry, I'm not very familiar with the reviewing process)?

File: src/rusthello/players/mod.rs (hunk 1 above)
Comment (NilSet) [inline src/rusthello/players/mod.rs]: Is there a style guide or a linter I should be using?

**Comment by ticki** [comment]
Ok, I'll merge then!

File: src/rusthello/players/mod.rs (hunk 1 above)
Comment (ticki) [inline src/rusthello/players/mod.rs]: Rustfmt is probably what comes nearest.

**Pull request merged**